newrelic/hooks/serverless_azure.py

```diff
@@ -0,0 +1,298 @@
+# Copyright 2010 New Relic, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import re
+import urllib.parse as urlparse
+
+from newrelic.api.application import application_instance
+from newrelic.api.transaction import current_transaction
+from newrelic.api.web_transaction import WebTransaction
+from newrelic.common.object_wrapper import wrap_function_wrapper
+
+
+# TODO: This should serve as a way to determine the trigger type.
+# Right now, we only support HTTP, so this function serves to activate
+# the application if not already registered with the collector as well
+# as determining if this invocation was a cold start or not.
+async def wrap_dispatcher__handle__invocation_request(wrapped, instance, args, kwargs):
+    def bind_params(request, *args, **kwargs):
+        return request
+
+    # Force default registration of the application instance
+    # instead of lazy registration upon the first request
+    app_name = os.environ.get("NEW_RELIC_APP_NAME", os.environ.get("WEBSITE_SITE_NAME", None))
```

Comment:
Wondering if we could this application activation logic into a separate function since it seems to be used a few times across wrappers.